src/Heightmap.cc

```diff
@@ -324,11 +325,14 @@ Errors Heightmap::Load(ElementPtr _sdf, const ParserConfig &_config)
     return errors;
   }
 
+  auto config = _config;
+  auto path = std::filesystem::path(this->dataPtr->filePath).parent_path();
```

Comment:
`filePath` may not be set if, for example, the object was constructed programmatically or the SDF input was a string. Can you guard against that?

Reply:
it should just return an empty string. I added a check nevertheless [443dfd2](https://github.com/gazebosim/sdformat/pull/1373/commits/443dfd200d7f0f341d42b10a280de58c235d39d9)